### PR TITLE
Update docs for SPNEGO configuration

### LINKED ISSUE #1909: Spnego JAAS file is not considered
#### **Issue**
CAS 4.2.4-SNAPSHOT

Setting cas.spnego.login.conf.file property in cas.properties is not considered (mark as null in log).
Same result if I declare a JcifsConfig bean in deployerConfigContext.
I have to report JAAS file through "java.security.auth.login.config" in JAVA_OPTS for spnego authentication works.
It was OK in CAS 4.0.x

I will provide an overlay tomorrow.

### PR DESCRIPTION
**Pull request by lsenecaux**
Closes #1909 and #1919

### CODE REVIEW

File: cas-server-documentation/installation/SPNEGO-Authentication.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ SPNEGO support is enabled by including the following dependency in the Maven WAR
 ```xml
 <dependency>
   <groupId>org.jasig.cas</groupId>
-  <artifactId>cas-server-support-spnego-webflow</artifactId>
```
Comment (mmoayyed) [inline cas-server-documentation/installation/SPNEGO-Authentication.md]: This is not right the change. The webflow module should be creating everything SWF needs. If it's not, we should look there and see what might be failing.

File: cas-server-documentation/installation/SPNEGO-Authentication.md [hunk 2]
```diff
@@ -94,19 +94,46 @@ zone.
 URL, e.g. `https://cas.example.com`.
 
 ### Webflow Configuration
+Create new action state in /WEB-INF/webflow/login-webflow.xml, just before `viewLoginForm` one.
```
Comment (mmoayyed) [inline cas-server-documentation/installation/SPNEGO-Authentication.md]: See above. All this should be unnecessary ideally.

**Comment by mmoayyed** [comment]
Thanks very much for the pull. I'll need to take a look at the webflow module and see why it's failing to create those SWF changes. Those should all be automated.

**Comment by lsenecaux** [comment]
Perhaps is it not instantiated before the viewLoginForm action state ?

**Comment by mmoayyed** [comment]
Dont think so. At least, it's not obvious to me why that would have any effect. Have you tried this line in your config?

```
Replace  `viewLoginForm` with `startSpnegoAuthenticate`, if any.
```

That's what the auto-configuration step does not do.

**Comment by lsenecaux** [comment]
Yes, I tried it, and there is a loop when I try to authenticate.
Then, I add the manually the 2 action states, remove the module, and it works

**Comment by mmoayyed** [comment]
OK. Strange. This definitely needs further review.

Could I ask you to try it based on the way the docs suggest, but turn up logging for org.springframework.webflow to at least DEBUG? That should further explain what SWF is actually doing.

**Comment by mmoayyed** [comment]
I'll run through this myself as well. Been a while since I have looked at this one.

**Comment by lsenecaux** [comment]
Sorry, but I could not.
I am not at office until September 5th, and I haven't got a dev machine at home.

**Comment by mmoayyed** [comment]
No worries. We can leave the pull open by then. At least for the time being, this seems like a decent solution until we get to the bottom of this. Thanks again.

**Comment by lsenecaux** [comment]
you are welcome

**Comment by mmoayyed** [comment]
To elaborate further, the confusing part to me is, I am not seeing how the flow would jump to `startSpnegoAuthenticate`. Even if you declared it before that `viewLoginForm`, nothing before that state navigates to `startSpnegoAuthenticate`. So we'll need better logs to figure this out. All in good time.

**Comment by lsenecaux** [comment]
I just followed the documentation for version 4.0 and I applied it to the 4.2

**Comment by mmoayyed** [comment]
Gotcha. When the time comes, please do share logs so we can spot the differences. I am more than glad to adjust the auto configuration logic to create those actions before that particular state, but I am more curious how it actually used to work in 4.0, and not having a proper test/dev environment to set this up makes that difficult. 

Appreciate your help.

**Comment by mmoayyed** [comment]
Merged out of band. Be sure to test again please when 4.2.4 comes out.

**Pull request closed without merging**